modules/ce_google_docs/src/Controller/GoogleDocsController.php

```diff
@@ -0,0 +1,190 @@
+
+/**
+ * Class GoogleDocsController.
+ */
+class GoogleDocsController extends ControllerBase {
+
+  /**
+   * To update a node.
+   *
+   * @param Symfony\Component\HttpFoundation\Request $request
+   *   HttpRequest with various parameters of a a node.
+   *
+   * @return Symfony\Component\HttpFoundation\JsonResponse
+   *   JsonResponse with Node's details.
+   */
+  public function update(Request $request) {
+    $config = \Drupal::config('ce_google_docs.settings');
+    if ($request->request->get('apiKey') != $config->get('api_key')) {
+      return new JsonResponse(['error' => 'Invalid API Key.']);
+    }
+
+    $body = $request->request->get('body');
+    if ($request->request->get('img_store') == 'drupal') {
+      preg_match_all('/< *img[^>]*src *= *["\']?([^"\']*)/i', $body, $result);
+      $result = $result[1];
+      foreach ($result as $src) {
+        $client = new Client();
+        $response = $client->request('GET', $src);
+        $name = str_replace(" ", "_", substr($response->getHeader('Content-Disposition')[0], 17, -1));
+        $file = system_retrieve_file($src, 'public://' . $name, TRUE, $replace = FILE_EXISTS_RENAME);
+        $new_src_url = explode($_SERVER['HTTP_HOST'], $file->url())[1];
+        $body = str_replace($src, $new_src_url, $body);
+      }
+    }
+
+    $node = Node::load($request->request->get('nid'));
+    $node->title = $request->request->get('title');
+    $node->body->summary = $request->request->get('summary');
+    $node->body->value = $body;
+    $node->save();
+
+    return new JsonResponse(['nid' => $node->id()]);
+  }
+
+  /**
+   * To save/publish a node.
+   *
+   * @param Symfony\Component\HttpFoundation\Request $request
+   *   HttpRequest with various parameters of a a node.
+   *
+   * @return Symfony\Component\HttpFoundation\JsonResponse
+   *   JsonResponse with Node's details.
+   */
+  public function publish(Request $request) {
+    $config = \Drupal::config('ce_google_docs.settings');
+    if ($request->request->get('apiKey') != $config->get('api_key')) {
+      return new JsonResponse(['error' => 'Invalid API Key.']);
+    }
+
+    $body = $request->request->get('body');
+    if ($request->request->get('img_store') == 'drupal') {
+      preg_match_all('/< *img[^>]*src *= *["\']?([^"\']*)/i', $body, $result);
+      $result = $result[1];
+      foreach ($result as $src) {
+        $client = new Client();
+        $response = $client->request('GET', $src);
+        $name = str_replace(" ", "_", substr($response->getHeader('Content-Disposition')[0], 17, -1));
+        $file = system_retrieve_file($src, 'public://' . $name, TRUE, $replace = FILE_EXISTS_RENAME);
+        $new_src_url = explode($_SERVER['HTTP_HOST'], $file->url())[1];
+        $body = str_replace($src, $new_src_url, $body);
+      }
+    }
+
+    $body_feild = '';
+    $fields = \Drupal::entityManager()->getFieldDefinitions('node', $request->request->get('type'));
+    foreach ($fields as $field => $value) {
```

Comment:
What if I have added multiple fields type like email number so how will you map them